geowrangler/grids.py

```diff
@@ -200,21 +202,23 @@ def __init__(
         self,
         zoom_level: int,  # Zoom level of tile. See: https://docs.microsoft.com/en-us/bingmaps/articles/bing-maps-tile-system for more info
         return_geometry: bool = True,  # If geometry should be returned. Setting this to false will only return quadkeys
+        add_xy_cols: bool = False,  # If quadkey should be converted to their xy values.
     ):
         self.zoom_level = zoom_level
         self.return_geometry = return_geometry
+        self.add_xy_cols = add_xy_cols
         self.tms = morecantile.tms.get("WebMercatorQuad")
 
     def tile_to_polygon(self, tile: morecantile.Tile):
-        """Concerts a tile to geometry"""
+        """Converts a tile to geometry"""
         return shape(self.tms.feature(tile)["geometry"])
 
     def get_tiles_for_polygon(
         self,
         polygon: Polygon,
         filter: bool = True,
     ):
-        """Get the interseting tiles with polygon for a zoom level. Polygon should be in EPSG:4326"""
+        """Get the intersecting tiles with polygon for a zoom level. Polygon should be in EPSG:4326"""
         x_min, y_min, x_max, y_max = polygon.bounds
         tiles = (
             (self.tms.quadkey(tile), self.tile_to_polygon(tile))
```

Comment:
I would prefer you extract xyz from tile object here rather than recomputing it. 

You can access the x,y,z from a tile via `tile.x`, `tile.y`, `tile.z`

You can update the following line to  

```python
   tiles = (
            (self.tms.quadkey(tile), self.tile_to_polygon(tile), tile.x, tile,y, tile.z) # add x, y, z
            for tile in self.tms.tiles(x_min, y_min, x_max, y_max, self.zoom_level)
        )
        # Return dict to make it easier to deduplicate
        if filter:
            tiles = {qk: (geom,x,y,z) for qk, geom, x, y, z,  in tiles if polygon.intersects(geom)} # upadte returned type to a tuple
        else:
            tiles = {qk:  (geom,x,y,z) for qk, geom, x, y, z in tiles} # update returned type to a tuple
        return tiles
 ```

later down you would replace
```python
quadkey, geom = zip(*((k, v) for k, v in tiles.items()))
```
with 
```python
quadkey, geom_xyz, = zip(*((k, v) for k, v in tiles.items()))
geom, x, y, z = zip(*geom_xyz)
```

then you can do
```python
df['x'] = list(x)
```

Reply:
Somewhat related but I'm also writing a [parallelized version of the filtering part ](https://github.com/thinkingmachines/geowrangler/blob/4b6292354c16b8676f868b7d69e505930730687b/geowrangler/grids.py#L342) because based on my profiling on bing tile gridding for large areas (e.g. Indonesia), this is where the process bottlenecked.